yb-voyager/cmd/exportData.go

```diff
@@ -798,6 +807,200 @@ func getInitialTableList() (map[string]string, []sqlname.NameTuple) {
+				return getNameTupleFromObjectName(obj)
+			})
+			partitionsToRootTableMap = msr.TargetRenameTablesMap
+		}
+
+	}
+
+	//guardrails around the table-list in case of re-run
+
+	//TODO: fix later this registered list also contains sequence names right now
+	//so in case some include/exclude table names are not proper we will display this registeredList as valid table names
+	registeredListObjNames, err := namereg.NameReg.GetRegisteredTableList()
+	if err != nil {
+		utils.ErrExit("error getting registered list in name registry: %v", err)
+	}
+
+	registeredList := lo.Map(registeredListObjNames, func(obj *sqlname.ObjectName, _ int) sqlname.NameTuple {
+		return getNameTupleFromObjectName(obj)
+	})
+
+	excludeTableList := extractTableListFromString(registeredList, source.ExcludeTableList, "exclude")
+	if len(excludeTableList) > 0 {
+		_, excludeTableList, err = addLeafPartitionsInTableList(excludeTableList, true)
+		if err != nil {
+			utils.ErrExit("adding leaf partititons to exclude table list: %s", err)
+		}
+	}
+	includeTableList := finalTableList
+	if source.TableList != "" {
+		includeTableList = extractTableListFromString(registeredList, source.TableList, "include")
+		_, includeTableList, err = addLeafPartitionsInTableList(includeTableList, true)
+		if err != nil {
+			utils.ErrExit("adding leaf partititons to include table list: %s", err)
+		}
+	}
+	filteredTableListFromFlags := sqlname.SetDifferenceNameTuples(includeTableList, excludeTableList)
+
+	//Reporting the guardrail msgs only on leaf tables to be consistent so filtering the root table from both the list
+	rootTables := make([]sqlname.NameTuple, 0)
+	for _, v := range partitionsToRootTableMap {
+		tuple, err := namereg.NameReg.LookupTableName(v)
+		if err != nil {
+			utils.ErrExit("look up failed for the table name: %s: %v", v, err)
+		}
+		rootTables = append(rootTables, tuple)
+	}
+
+	filteredListWithoutRootTable := lo.Filter(filteredTableListFromFlags, func(t sqlname.NameTuple, _ int) bool {
+		return !lo.ContainsBy(rootTables, func(root sqlname.NameTuple) bool {
+			return t.Equals(root)
+		})
+	})
+
+	// Finding all the partitions of all root tables part of table-list, and report if there any new partitions added
+	updatedPartitionsToRootTableMap, _, err := addLeafPartitionsInTableList(rootTables, true)
+	if err != nil {
+		utils.ErrExit("getting updated partitions to root table mapping: %s", err)
+	}
+
+	newLeafTables := make(map[string][]string)
+	for leaf, value := range updatedPartitionsToRootTableMap {
+		if !lo.ContainsBy(registeredList, func(tbl sqlname.NameTuple) bool {
+			return tbl.AsQualifiedCatalogName() == leaf
+		}) {
+			//If this leaf table is not registered in name registry then it is a newly added leaf tables
+			newLeafTables[value] = append(newLeafTables[value], leaf)
+		}
+	}
+
+	if len(lo.Keys(newLeafTables)) > 0 {
+		utils.PrintAndLog("Detected new partition tables for the following partitioned tables. These will not be considered during migration:")
```

Comment:
Should we prompt?

Reply:
Not sure if a prompt is required, I think there is nothing we want users to do here right? so this might just FYI?